src/evaluate/evaluate.ts

```diff
@@ -77,9 +90,9 @@ export class Evaluator {
     return new ConstructReference(logicalId, value as Construct);
   }
 
-  public evaluate(x: TypedTemplateExpression): any {
+  public evaluate(x: TypedTemplateExpression | TemplateExpression): any {
```

Comment:
Is there a way we can resolve the `TemplateExpression` for the output value to a `TypedTemplateExpression`?

I don't like the implications of giving up the typed context here, there are a lot of assumptions based on the fact that we only deal with typed expressions here.

Reply:
Yea sure I can convert it to `TypedTemplateExpression`.

Would you mind elaborating what are the assumptions? I added b/c I found `TemplateExpression` covers a subset of value of `TypedTemplateExpression`.

Reply:
That's a good question and you're probably right that there a little practical differenves. 

A few thoughts on this:
- If `TemplateExpression` indeed are a true subset of `TypedTemplateExpression` then we should be able to setup the types in a way that `TemplateExpression` is accepted where `TypedTemplateExpression` is expected
- There's some semantic value to 'see' that we are now in typed land, vs. untyped land.
- One thing the type resolution does is converting `ObjectLiterals` to method calls etc. So while `TypedTemplateExpression` still accepts object literals, we now know that they only appear in the places we expect them.

Hope that makes sense.